homeassistant/components/hausbus/__init__.py

```diff
@@ -0,0 +1,46 @@
+"""The Haus-Bus integration."""
+from __future__ import annotations
+
+import logging
+from typing import cast
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .gateway import HausbusGateway
+
+# Only support light platform on the initial version. Additional platforms to follow
+PLATFORMS: list[Platform] = [Platform.LIGHT]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Haus-Bus from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    gateway = HausbusGateway(hass, entry)
+    hass.data[DOMAIN][entry.entry_id] = gateway
```

Comment:
Instead of this, implement https://developers.home-assistant.io/blog/2024/04/30/store-runtime-data-inside-config-entry